tensorflow/compiler/xla/service/gpu/gpu_layout_assignment.cc

```diff
@@ -57,6 +57,11 @@ HeuristicLayoutAssignment(const HloInstruction* instr,
       std::make_tuple(DataLayout::kBatchYXDepth, FilterLayout::kOutputYXInput,
                       DataLayout::kBatchYXDepth);
 
+  // Integer convolution must use NHWC.
+  if (primitive_util::IsIntegralType(instr->operand(0)->shape().element_type())) {
```

Comment:
CuDNN supports integer convolution since compute capability 6.1, prior to Volta.
Integer convolution requires NHWC since then. 
Do you ask how to handle earlier chips that do not support integer convolution?  If so, I intend to leave the sanity check to the executor.